src/compiler/utilities.ts

```diff
@@ -7119,6 +7119,39 @@ namespace ts {
         }
     }
 
+    export function nodeIsFirstNodeAtOrAfterPosition(sourceFile: SourceFile, node: Node, position: number): boolean {
+        if (node.pos === position) return true;
+        if (node.pos < position) return false;
+        let current: Node = sourceFile;
+        let next: Node | undefined;
+        const getContainingChild = (child: Node) => {
+            if (child.pos <= position && (position < child.end || (position === child.end && (child.kind === SyntaxKind.EndOfFileToken)))) {
+                return child;
+            }
+            else if (!next && child.pos > position) {
+                next = child;
+            }
+        };
+        while (true) {
+            const child = isSourceFileJS(sourceFile) && hasJSDocNodes(current) && forEach(current.jsDoc, getContainingChild) || forEachChild(current, getContainingChild);

```

Comment:
Is this doing a recursive walk of a Node's entire subtree? That sounds expensive for something could possibly call between every list element.

Reply:
This only ever happens when printing nodes from `TextChanges`.

Reply:
Nonetheless, I reworked to make it start at the nearest possible ancestor instead of the source file, and skipped it altogether in cases where neither list element went through a transformation (which is the only way they could be reordered or reparented, I think).